social/twitter/27-twitter.js

```diff
@@ -198,111 +246,280 @@ module.exports = function(RED) {
+                                streamConnect(++retryAttempt);
+                                node.status({ fill: "red", shape: "ring", text: RED._("twitter.errors") });
+                            }, 2 ** retryAttempt)
+                        }
+                        if (node.restart) {
+                            node.tout = setTimeout(function () { setupStream() }, retry);
+                        }
+                    }).on('limit', limit => {
+                        //node.status({fill:"grey", shape:"dot", text:RED._("twitter.errors.limitrate")});
+                        node.status({ fill: "grey", shape: "dot", text: (tags || " ") });
+                    }).on('destroy', function (response) {
+                        twitterRateTimeout = Date.now() + 15000;
+                        if (node.restart) {
+                            node.status({ fill: "red", shape: "dot", text: " " });
+                            node.warn(RED._("twitter.errors.unexpectedend"));
+                            node.tout = setTimeout(function () { setupStream() }, 15000);
+                        }
+                    });
+
+                    return stream;
+
+                }
+
                 try {
-                    var thing = 'statuses/filter';
                     var tags = node.tags;
                     var st = { track: [tags] };
-
-                    var setupStream = function() {
+                    var setupStream = function () {
                         if (node.restart) {
-                            node.status({fill:"green", shape:"dot", text:(tags||" ")});
-                            twit.stream(thing, st, function(stream) {
-                                //console.log("ST",st);
+                            (async () => {
+                                let currentRules;
+                                console.warn = ({ topic: "warning", payload: node.restart })
+                                try {
+                                    // Gets the complete list of rules currently applied to the stream
+                                    currentRules = await getAllRules();
+                                    // Delete all rules. Comment the line below if you want to keep your existing rules.
+                                    await deleteAllRules(currentRules);
+                                    // Add rules to the stream. Comment the line below if you don't want to add new rules.
+                                    await setRules();
+                                } catch (e) {
+                                    console.error(e);
+                                    process.exit(1);
+                                }
+                                // Listen to the stream.
+                                // node.status({fill:"green", shape:"dot", text:(node.tags||" ")});
+                                console.log("Twitter API is steraming public tweets with search term " + node.tags || " ");
```

Comment:
Love the project to upgrade to Twitter 2.0.  Would be great to keep the logging consistent with the old node, and avoid console.log() throughout, so node.debug maybe here?

Reply:
Good suggestion, I will change it.